volume/mounts/linux_parser.go

```diff
@@ -1,3 +1,5 @@
+// +build !windows
```

Comment:
Nit as it's previously like this. But I would have thought this file would be parser_linux.go to use built-in tagging by file extension. Similarly parser_windows.go which wouldn't need a tag, and then parser_lcow.go which would need a tag. That said, the lcow parser is so small, there's no reason its contents couldn't be in parser_windows.go too.

Reply:
I agree. I recall some discussion around more code sharing between parsers, relating to https://github.com/moby/moby/pull/34258, but I cannot find the comments now. Do you remember anything about that?

@simonferquel, do you recall why the files are `<os>_parser.go` instead of `parser_<os>.go`?

Reply:
IIRC there were 2 aspects of that:
- we wanted to be able to ultimately improve code sharing between OSes (having parts of lcow support leverage code from the Linux parsing/validation rules)
- as the Package is not internal, we saw potential opportunities for 3rd party software to be able to vendor this code and be able to validate mount expressions for both windows and linux (exemple I have in mind, is the Docker for Windows connection proxy that could invoke the Linux parser when active backend is MobyLinuxVM and Windows parser when active backend is Windows Daemon)

Reply:
And we also anticipated that the kind of multi-platform support at runtime introduced by LCOW could eventually arrive in other forms (like Linux being able to run Windows containers in the future trough KVM or something similar), and that we should try to avoid platform-specific code for parsing

Reply:
@simonferquel Thanks.

If we want to keep that flexibility, I can fix this differently by adding a wrapper function to get the OS version information. It would never be called for Linux. but the Windows parser, running on Linux, would need an alternate way of querying the Windows build. That seems outside the scope of this package, unless we provide a function to initialize the Windows OS version.

Alternatively, we can drop the build and directory checks altogether. This may result in a slightly worse user experience, but will work in all supported cases, and simplify this change.

What do you think?

Reply:
I have not a strong advice about it, but if we could keep things simple (e.g.: only 1 validation path for windows) I think it would be less of a burden to handle. Maybe if we can detect runtime errors related to file mounts, we could have almost the same ux as today, but the error would be caught later in the process

Reply:
I do not mind doing it with this change, but the questions are: do we want to keep them all building for all platforms, or make it easy to do so in the future, per Simon's comments above? Perhaps I should just make this change trivial, and remove the directory check entirely.

Reply:
I remember now also one thing that could make sense: if 3rd party code (including Docker CLI) needs to use code from this package, it might need to parse mounts that are targeting an engine that has not the same platform / os / version. This was another reason to make every parser available on every platform combination.
@johnstep @thaJeztah does it make sense ?

Reply:
Ah, right, yes.

@johnstep I see there's some other review comments to be addressed as well; could you update the pull request?

Reply:
This is a big if, and I think not being done now.

One of the issues with the client is also not knowing what platform the daemon is on and as such which parser to even use.
We could possibly determine at least what the container platform will be from looking at the image manifest, but this does not mean the host is running that platform (e.g. LCOW).

Reply:
Discussing with @cpuguy83 in the maintainers meeting, and it looks like a better plan to put them behind build-flags (or `_linux.go`), and not use these in the client.